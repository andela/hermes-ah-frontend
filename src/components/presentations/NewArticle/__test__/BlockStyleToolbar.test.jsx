import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import HeaderDropdown from '../BlockStyles/BlockStyleToolbar';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const store = mockStore({});

const ArticleHeader = (
  <Provider store={store}>
    <HeaderDropdown editorState="article" onToggle={jest.fn()} />
  </Provider>
);

describe('NewArticle button component', () => {
  it('should match snapshot', () => {
    const wrap = ArticleHeader;
    expect(wrap).toMatchSnapshot();
  });

  it('should render without crashing', () => {
    const wrapper = shallow(ArticleHeader);
    expect(wrapper.find('span'));
  });
});
