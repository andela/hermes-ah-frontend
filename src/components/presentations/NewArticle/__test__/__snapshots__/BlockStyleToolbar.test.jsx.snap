// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewArticle button component should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BlockStyleToolbar
    editorState="article"
    onToggle={[MockFunction]}
  />
</Provider>
`;
